test/runtests.jl

```diff
@@ -217,6 +220,79 @@ using Test
+        mgmmw = IsotropicMultiGMM(Dict(:Hydrophobe   => IsotropicGMM([IsotropicGaussian([ 0.0, 0.0], 2.0, 1.0)]),
+                                       :Donor        => IsotropicGMM([IsotropicGaussian([ 0.0, 0.0], 1.5, 1.0)]),
+                                       :PosIonizable => IsotropicGMM([IsotropicGaussian([ 0.0, 0.0], 4.0, 1.0)])))
+        mgmmx = IsotropicMultiGMM(Dict(:Hydrophobe   => IsotropicGMM([IsotropicGaussian([ 1.0, 0.0], 2.0, 1.0)])))
+        mgmmy = IsotropicMultiGMM(Dict(:Acceptor     => IsotropicGMM([IsotropicGaussian([-1.0, 1.0], 1.5, 1.0)])))
+        mgmmz = IsotropicMultiGMM(Dict(:NegIonizable => IsotropicGMM([IsotropicGaussian([-1.0,-1.0], 4.0, 1.0)])))
+        forces = forcecomponents([mgmmw, mgmmx, mgmmy, mgmmz], interactions, [1])
+        @test length(forces) == 1
+        @test all(f -> norm(sum(f; dims=2)) > 0.1, forces)   # non-zero force
+        f = only(forces)
+        F = f' * f
+        # Test that each force is in the expected direction
+        @test f[:, 1] ≈ ([ 1, 0] \ f[:, 1]) * [ 1, 0]
+        @test f[:, 2] ≈ ([-1, 1] \ f[:, 2]) * [-1, 1]
+        @test f[:, 3] ≈ ([-1,-1] \ f[:, 3]) * [-1,-1]
+        w = optimize_weights(forces)
+        @test all(>=(0), w)
+        @test sum(w) ≈ 1
+        weighted_interactions = [key => w0*wi for ((key, w0), wi) in zip(interactions, w)]
+        wforces = forcecomponents([mgmmw, mgmmx, mgmmy, mgmmz], weighted_interactions, [1])
+        @test all(f .* w' ≈ wf for (f, wf) in zip(forces, wforces))
+        @test all(f -> norm(sum(f; dims=2)) < 1e-6, wforces)   # net-zero force
+        wF = sum(f' * f for f in wforces)
+        @test abs(ones(3)'*(wF*ones(3))) < 1e-12
+        # wnew = optimize_weights(wforces)   # HiGHS gets stuck, possibly the solution is degenerate??
+        # @test all(≈(1/length(interactions)), wnew)
+
+        interactions = [(:Steric, :Steric) => 1, (:Hydrophobe, :Hydrophobe) => -1, (:Donor, :Acceptor) => -1]  # drop the ionic
+        opsd = read("AF-P15409-F1-model_v4.pdb", PDBFile)
+        opsd_tms = [37:61, 74:96, 111:133, 153:173, 203:224, 253:274, 287:308]
+        tm_res = inward_tm_residues(opsd, opsd_tms)
+        tm_idxs = reduce(vcat, [tm[idx] for (tm, idx) in zip(opsd_tms, tm_res)])
+        for combined in (false, true)
+            forces = forcecomponents(opsd, interactions, tm_idxs; combined)
+            w = optimize_weights(forces)
+            # Exact force balance is presumably impossible to achieve in this case.
+            # For a weaker test, check that weighted interactions need no additional weighting.
+            weighted_interactions = [key => w0*wi for ((key, w0), wi) in zip(interactions, w)]
+            wforces = forcecomponents(opsd, weighted_interactions, tm_idxs; combined)
+            @test all(f .* w' ≈ wf for (f, wf) in zip(forces, wforces))
+            wnew = optimize_weights(wforces)
+            @test_broken all(≈(1/length(interactions)), wnew)
```

Comment:
I'm puzzled about why this isn't passing. I worry there's some bug that is revealed by a complex example and isn't caught by the simple examples above.

Reply:
OK, it looks like it's simply a very, very flat basin:
![image](https://github.com/HolyLab/GPCRAnalysis.jl/assets/1525481/e56e3372-08d3-4085-9925-a1a528d8b672)